packages/node_modules/@webex/webex-core/src/lib/services/service-url.js

```diff
@@ -56,10 +56,24 @@ const ServiceUrl = AmpState.extend({
       return this.defaultUrl;
     }
 
-    const filteredHosts = clusterId ?
+    let filteredHosts = clusterId ?
       this.hosts.filter((host) => host.id === clusterId) :
       this.hosts.filter((host) => host.homeCluster);
 
+    const aliveHosts = filteredHosts.filter(
+      (host) => !host.failed
+    );
+
+    filteredHosts = (aliveHosts.length === 0) ?
+      filteredHosts.map(
+        (host) => {
+          /* eslint-disable-next-line no-param-reassign */
+          host.failed = false;
```

Comment:
You could remove the eslint disable by creating a new object and returning:
```suggestion
          const aliveHost = {...host, failed: false};
```

Reply:
If the idea is to update those hosts in the `this.hosts` array directly, it might be better to make that a specific function since this function is a getter and you don't really want side effects in a getter.

Reply:
Properly implementing this would require the creation of a `ServiceHost` class, as there isn't a way to provide mutability while also avoiding param reassign on nested objects. This could be done as a part of the planned refactor of this plugin. Examples:

This would create mismatched data:
```js
class ServiceUrl {
  ...
  resetHosts() {
    const preHosts = this.hosts;

    this.hosts = this.hosts.map(
      (host) => ({...host, failed: false})
    );

    if (this.hosts !== preHosts) {
      throw new Error('bad mutability! cannot do!');
    }
}
```